x/crosschain/genesis.go

```diff
@@ -46,7 +46,7 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState)
 
 	// Set all the send
 	for _, elem := range genState.CrossChainTxs {
-		k.SetCrossChainTx(ctx, *elem)
+		k.SetCctxAndNonceToCctxAndInTxHashToCctx(ctx, *elem)
```

Comment:
Why are we removing the SetCCTX here? 
Also init genesis is only called once when the module is added , it is not called for at upgrades